### PR TITLE
employment duration filter

### LINKED ISSUE #2768: Employment duration -  New filter in request page.
[Wireframe](https://www.figma.com/proto/XS4Ag6GWcgdq2dBlLzBkay/Admin-v3?page-id=0%3A1&node-id=949%3A5635&viewport=-1637%2C-2263%2C0.73&scaling=min-zoom&starting-point-node-id=949%3A5635&show-proto-sidebar=1)

Add new filter to the request talent page called "Employment duration". This should be made to the NEW SearchForm.

Any Duration is equivalent to leaving this filter empty.

image

## Acceptance Criteria

- [ ] These inputs are connected to the api query
- [ ] UI is done

### PR DESCRIPTION
**Pull request by vd1992**
resolves #2768 

to connect the filter to API, there needs to well, be something to connect it to so I'm playing a bit stretchy with the issue scope
I don't think it is that much extra work to add the backend tweaks for it to connect, needed so the frontend filter is actually test-able

I think things are in order? Not sure if anything else is missing
Test: play around with /search and ensure the database actions match expectations 

labels are a bit funny, two options are supposed to do nothing but giving them the same value throws alerts

### CODE REVIEW

**Comment by vd1992** [comment]
reference for added filter radiogroup

image

**Comment by vd1992** [comment]
playing around with SQL queries, and learning how to make them!
so far, what appears on /search with duration = short term yields 1
this query in adminer also yields 1 so... I think things look to be working?

```
SELECT COUNT(*)
FROM "users"
JOIN "pool_candidates" ON ("pool_candidates"."user_id" = "users"."id")
WHERE "pool_candidates"."user_id" = "users"."id"
AND "pool_candidates"."pool_id" = '1dcfd143-083c-4a9a-a8da-88f7876ebb99'
AND "pool_candidates"."pool_candidate_status" = 'AVAILABLE'
AND "pool_candidates"."expiry_date" >= '2022-06-22'
AND "users"."would_accept_temporary" = '1'
```

File: api/database/migrations/2022_06_22_170914_update_pool_candidatate_filter_employment_duration.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('pool_candidate_filters', function (Blueprint $table) {
+            $table->boolean('would_accept_temporary')->nullable();
```
Comment (JamesHuf) [inline api/database/migrations/2022_06_22_170914_update_pool_candidatate_filter_employment_duration.php:17]: While the query itself works fine as a boolean we should probably save this as an enum/string so we can actually tell whether a user selected 'any duration' or 'permanent'.

**Review by JamesHuf — CHANGES_REQUESTED** [review]
Another great PR! There's just one change I think we'll want.

File: api/database/migrations/2022_06_22_170914_update_pool_candidatate_filter_employment_duration.php (hunk 1 above)
Comment (vd1992) [reply api/database/migrations/2022_06_22_170914_update_pool_candidatate_filter_employment_duration.php]: I realized I didn't even need to add that into this PR, I didn't need any API/DB changes.
That part is stealing from #3106 without real reason so just going to undo the migration bit 😬  🤣

Comment (JamesHuf) [reply api/database/migrations/2022_06_22_170914_update_pool_candidatate_filter_employment_duration.php]: Sounds good!

**Review by JamesHuf — APPROVED** [review]
Everything looks great now!

**Pull request merged**